lensless/utils/dataset.py

```diff
@@ -1599,22 +1625,45 @@ def _get_images_pair(self, idx):
                 shape=(self.alignment["height"], self.alignment["width"], 3),
                 interpolation=cv2.INTER_NEAREST,
             )
+            background = (
+                resize(
+                    background_np,
+                    shape=(self.alignment["height"], self.alignment["width"], 3),
+                    interpolation=cv2.INTER_NEAREST,
+                )
+                if not None
+                else None
+            )
         elif self.display_res is not None:
             lensed = resize(
                 lensed_np, shape=(*self.display_res, 3), interpolation=cv2.INTER_NEAREST
             )
+            background = (
+                resize(background_np, shape=(*self.display_res, 3), interpolation=cv2.INTER_NEAREST)
+                if not None
+                else None
+            )
         elif self.downsample_lensed != 1.0:
             lensed = resize(
                 lensed_np,
                 factor=1 / self.downsample_lensed,
                 interpolation=cv2.INTER_NEAREST,
             )
+            background = (
+                resize(
+                    background_np,
+                    factor=1 / self.downsample_lensed,
+                    interpolation=cv2.INTER_NEAREST,
+                )
+                if not None
+                else None
+            )
 
-        return lensless, lensed
+        return lensless, lensed, background if background is not None else None
 
     def __getitem__(self, idx):
-        lensless, lensed = super().__getitem__(idx)
```

Comment:
Because you changed this line, you don't have torch arrays and the missing dimension